daemon/cluster/executor/container/adapter.go

```diff
@@ -65,17 +66,44 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 	}()
 
 	dec := json.NewDecoder(pr)
+	dec.UseNumber()
```

Comment:
Without `UseNumber`, the progress information gets unmarshalled to floats, and gets printed in scientific notation by logrus.

https://github.com/docker/swarmkit/pull/1352#discussion_r74291601